### PR TITLE
Icon: split background icon base64 encodings to new module #239

### PR DESCRIPTION
**Pull request by ianmcburnie**
# Issue #239

## Goal(s)

1. Split base64 icon encodings into separate module in a backwards compatible way

## Changes

1. Added new `@ebay/skin/base64` module as a dependency of @ebay/skin/icon
1. Created a new Lasso flag to opt out of the base64 module dependency (in future we can change to opt-in)
1. Slight refactoring of mixins (in backwards compatible way) - but this can be further cleaned up in v7

## Non-Goals

1. Due to the new base64 module being included by default, I decided not to add a link in the left nav for now.

## Screenshots

See that the majority of the file weight has moved to the new dist files:

screen shot 2018-12-05 at 3 01 57 pm

Updated `icon.browser.json`

screen shot 2018-12-05 at 3 09 36 pm

### CODE REVIEW

File: src/less/less/ds6/variables.less [hunk 1]
```diff
@@ -317,7 +317,80 @@
...
-
+@ds6-icon-arrow-move-base64:
+'PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHhtbG5zOnhsaW5rPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5L3hsaW5rIiB3aWR0aD0iMTIiIGhlaWdodD0iMTIiPjxkZWZzPjxwYXRoIGlkPSJhIiBkPSJNMy4yOTcgMS45MjRoNy42OWMuNTQ2LS4wMDEuOTYyLS40ODEuOTYyLS45NjFzLS40NDgtLjk2Ni0uOTYxLS45NjNDMTAuNDc1LjAwNCAwIDAgMCAwcy4wMTcgMTAuMzQgMCAxMC45ODljLS4wMTUuNjQ3LjQ3Ljk2MS45NjMuOTYxLjQ5IDAgLjk0Ny0uMzI1Ljk2MS0uOTYxLjAxNC0uNjM1IDAtNy42OTIgMC03LjY5Mmw4LjM3NyA4LjM3OWMuNDgzLjQ4MiAxLjA1OC4zNyAxLjQwMS4wMjcuMzQ0LS4zNDMuNDU4LS45MTYtLjAyNy0xLjQwMUwzLjI5NyAxLjkyNHoiLz48L2RlZnM+PHVzZSBmaWxsPSIjMDAwIiB4bGluazpocmVmPSIjYSIgZmlsbC1ydWxlPSJldmVub2RkIi8+PC9zdmc+';
```
Comment (seangates) [inline src/less/less/ds6/variables.less]: Why are there extra new lines before these new strings?

Comment (ianmcburnie) [reply src/less/less/ds6/variables.less]: Oh, that's weird. Didn't show up in my editor. Will fix those.

File: icon.browser.json [hunk 2]
```diff
@@ -1,5 +1,11 @@
 {
     "dependencies": [
+        {
```
Comment (DylanPiercey) [inline icon.browser.json]: Would it make more sense to just have two entry points for icons? One for background svgs and one for inline? I know we can import `@ebay/skin/base64` manually but what benefit are we getting from this flag?

Comment (ianmcburnie) [reply icon.browser.json]: We could do that for v7 when we can introduce breaking changes, but for this minor version upgrade I'm trying to keep things backwards compatible for current apps. Or maybe I'm missing something?

Comment (ianmcburnie) [reply icon.browser.json]: I do wonder if I could have `@ebay/skin/icon/base64` and `@ebay/skin/icon/svg` - and then `@ebay/skin/icon` just brings in both of them, thus retaining backwards compatibility.

Comment (seangates) [reply icon.browser.json]: Yeah, then a developer could be granular, and avoid the size penalty, by pointing just to `@ebay/skin/icon/inline` without needing to have a flag. I like this line of thinking.

Comment (DylanPiercey) [reply icon.browser.json]: @ianmcburnie I like that idea.

**Comment by ianmcburnie** [comment]
Closing in favor of #471

**Pull request closed without merging**